src/main/scala/ProducePngPyramid.scala

```diff
@@ -0,0 +1,114 @@
+      header =
+        "Generates a raster image layer of global world pop values masked against buffered roads",
+      main = {
+        val inputOrcPathOpt = Opts.option[String](
+          "inputOrcPath",
+          help = "The S3 path to the input ORC file to read geometries from")
+        val outputPathOpt = Opts.option[String](
+          "outputPath",
+          help = "The S3 path that the tile pyramid should be written to")
+
+        (inputOrcPathOpt, outputPathOpt).mapN {
+          (inputOrcPath, outputPath) =>
+            val conf =
+              new SparkConf()
+                .setIfMissing("spark.master", "local[*]")
+                .setAppName("World Pop Image Pyramid")
+                .set("spark.serializer", classOf[KryoSerializer].getName)
+                .set("spark.kryo.registrator", classOf[KryoRegistrator].getName)
+                .set("spark.executor.memory", "2g")
+                .set("spark.driver.memory", "2g")
+
+            implicit val ss =
+              SparkSession.builder.config(conf).enableHiveSupport.getOrCreate
+            // val sqlContext = ss.sqlContext
+            ss.withJTS
+
+            // TODO: Parameterize
+            val partitioner = SpatialPartitioner(64)
+
+            val osmRoadsDF: DataFrame = ss.read
+              .schema(BufferedRoadsSchema())
+              .orc(inputOrcPath)
+
+            val countryList = CountryDirectory.countries.map(_._2)  // TODO: Replace with full list?
+            val formatter =
+              CountryFormatter(ss.sparkContext, countryList.toList)
+            val countriesRDD: MultibandTileLayerRDD[SpatialKey] =
+              formatter.readCountries
+            val md: TileLayerMetadata[SpatialKey] = countriesRDD.metadata
+            val transform = md.mapTransform
+
+            val osmRoadsRdd: RDD[Feature[JTSGeometry, String]] = osmRoadsDF.rdd.map { geomRow =>
+              val geom = geomRow.getAs[JTSGeometry]("bufferedGeom")
+              val name = geomRow.getAs[String]("countryName")
+              Feature(geom, name)
+            }
+
+            val clippedGeoms: RDD[(SpatialKey, Feature[Geometry, String])] =
+              ClipToGrid(osmRoadsRdd, md.layout)
+
+            val groupedClippedGeoms: RDD[(SpatialKey, Iterable[Feature[Geometry, String]])] =
+              clippedGeoms.groupByKey(partitioner)
+
+            val joinedRdd: RDD[(SpatialKey, (MultibandTile, Iterable[Feature[Geometry, String]]))] =
+              countriesRDD.join(groupedClippedGeoms, partitioner)
+
+            val options: Rasterizer.Options = Rasterizer.Options(true, PixelIsArea)
+
+            val maskedRdd: RDD[(SpatialKey, Tile)] = joinedRdd.map {
+              case (spatialKey, (tile, features)) => {
+                val unionedGeom: JTSGeometry = features.map(_.geom).reduce { _ union _ }
+                val tileExtent: Extent = transform(spatialKey)
+                // TODO: Is retrieving only the first band correct?
```

Comment:
I believe it's fine for us to retrieve the first band here as the original data is a Singleband GeoTiff. It just happens to be read in as a MultibandTile for simplicity.